consensus/src/round_manager.rs

```diff
@@ -1353,6 +1353,14 @@ impl RoundManager {
             .with_label_values(&["order_vote_qc"])
             .observe(start.elapsed().as_secs_f64());
 
+        self.block_store
+            .sync_to_highest_quorum_cert(
+                order_vote_msg.quorum_cert().clone(),
+                self.block_store.sync_info().highest_commit_cert().clone(),
```

Comment:
this doesn't work, here it uses the local highest commit cert to check if it needs to sync, the function returns Ok immediately.

I think the simplest way is to check if NeedFetchResult::QCBlockExist before inserting and discard the order vote if we don't have the block locally.

Reply:
> discard the order vote if we don't have the block locally.

So, the idea is that after we discard order votes for a block not in block store, the subsequent proposal or vote will have a syncinfo that triggers the state sync?

Reply:
Nodes receive order votes before receiving the block far too often. Instead of discarding every such order vote, can I discard them only if the local highest_commit_cert is for a block more than let's say 7 rounds older than the order vote?

Reply:
i don’t think we want to fetch 7 blocks individually upon receiving order votes, we can still buffer the order vote that within 7 rounds but discard if there’s still no block after order qc formed

Reply:
Made the changes we discussed. Please take another look when you are free.